addon/index.ts

```diff
@@ -342,3 +342,30 @@ export const keepLatestTaskGroup = createDecorator(
 export const enqueueTaskGroup = createDecorator(createTaskGroupFromDescriptor, {
   enqueue: true
 });
+
+// @private
+function getEntries(object: object) {
+  const keys = Object.keys(object);
+  let i = keys.length;
+  const resultingArray = new Array(i); // preallocate the Array
+
+  while (i--) {
+    const key = keys[i];
+    const newEntry = [key, object[key]];
```

Comment:
I'm really new to Typescript, and I just can't figure this line out. The following error is noted:

```
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.  No index signature with a parameter of type 'string' was found on type '{}'.
```

Reply:
If you change the type definition of the `object` argument to `Record<string, any>` it should work.

You could also use `{[key: string]: any}` as both tell TS to allow access to the object keys by using an index.

Reply:
Is the `while` loop a performance optimization?
Otherwise you could neatly solve this with a `.map`:

```ts
export const objectEntries: typeof Object.entries = <T>(object: {
  [key: string]: T;
}) => Object.keys(object).map(key => [key, object[key]] as [string, T]);
```

Reply:
I ended up going with @buschtoens solution, seemed tighter to me.